File: client-app/src/app/models/Post.ts
import { IErrorResponse } from './Errors';

export interface IPost {
   id: string;
   displayName: string;
   userName: string;
   photoUrl: string;
   title: string;
   content: string;
   postedOn: string;
}

export interface IPagination {
   currentPage: number;
   itemsPerPage: number;
   totalItems: number;
   totalPages: number;
}

export interface IPostState {
   posts: IPost[];
   post: IPost | null;
   error: IErrorResponse | null;
   pagination: IPagination | null;
}

File: CleanArchitecture.Domain/Entities/Post.cs
using System;
using CleanArchitecture.Domain.Common;

namespace CleanArchitecture.Domain.Entities
{
   public class Post : AuditableEntity
   {
      public Guid Id { get; set; }
      public string DisplayName { get; set; }
      public string UserName { get; set; }
      public string PhotoUrl { get; set; }
      public string Title { get; set; }
      public string Content { get; set; }
   }
}